dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.Planning.Action;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+// ReSharper disable once CheckNamespace - Using NS of Plan
+namespace Microsoft.SemanticKernel.Planning;
+
+public sealed class ActionPlanner
+{
+    private const string StopSequence = "#END-OF-PLAN";
+    private const string SkillName = "basicPlanner";
+
+    // Planner semantic function
+    private readonly ISKFunction _plannerFunction;
+
+    // Context used to access the list of functions in the kernel
+    private readonly SKContext _context;
+    private readonly IKernel _kernel;
+
+    // TODO: allow to inject skill store
+    /// <summary>
+    /// Initialize a new instance of the <see cref="ActionPlanner"/> class.
+    /// </summary>
+    /// <param name="kernel">The semantic kernel instance.</param>
+    /// <param name="prompt">Optional prompt override</param>
+    public ActionPlanner(
+        IKernel kernel,
+        string? prompt = null)
+    {
+        Verify.NotNull(kernel, "The planner requires a non-null kernel instance");
+
+        string promptTemplate = prompt ?? EmbeddedResource.Read("skprompt.txt");
+
+        this._plannerFunction = kernel.CreateSemanticFunction(
+            skillName: SkillName,
+            promptTemplate: promptTemplate,
+            maxTokens: 1024,
+            stopSequences: new[] { StopSequence });
+
+        kernel.ImportSkill(this, skillName: SkillName);
+
+        this._kernel = kernel;
+        this._context = kernel.CreateNewContext();
+    }
+
+    public async Task<Plan> CreatePlanAsync(string goal)
```

Comment:
Planning on adding unit or integration tests in this PR or a follow up?

Reply:
https://github.com/orgs/microsoft/projects/811/views/1?pane=issue&itemId=26713338